# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xdm/startwm.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

More flexible root application startup.

--- xdm/config/Xsetup_0.orig	Wed Apr 27 09:20:03 1994
+++ xdm/config/Xsetup_0	Thu Dec 28 15:23:15 2000
@@ -1,3 +1,4 @@
 #!/bin/sh
 # $XConsortium: Xsetup_0,v 1.3 93/09/28 14:30:31 gildea Exp $
 xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+xsetroot -solid black
--- xdm-0.99.3/config/Xsession.cpp.orig	2005-11-28 00:19:46.944477270 +0100
+++ xdm-0.99.3/config/Xsession.cpp	2005-11-28 00:23:55.249879534 +0100
@@ -38,23 +38,15 @@
 	if [ -r "$resources" ]; then
 		BINDIR/xrdb -load "$resources"
 	fi
-#if defined(__SCO__) || defined(__UNIXWARE__)
-        [ -r /etc/default/xdesktops ] && {
-                . /etc/default/xdesktops
-        }
 
-        [ -r /etc/default/xdm ] && {
-                . /etc/default/xdm
-        }
+	# we need at least the PATH /opt/*/bin expansion ...
+	[ -f /etc/profile ] && . /etc/profile
 
-        XCOMM Allow the user to over-ride the system default desktop
-        [ -r $HOME/.xdmdesktop ] && {
-                . $HOME/.xdmdesktop
-        }
+	WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+	[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+	[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+	[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+	xsetroot -solid darkcyan ; exec twm ; exec xterm
 
-        [ -n "$XDESKTOP" ] && {
-                exec `eval $XDESKTOP`
-        }
-#endif
 	exec BINDIR/xsm
 fi
